backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/impl/HUPickingSlotBL.java

```diff
@@ -569,6 +571,15 @@ public void releasePickingSlotIfPossible(final I_M_PickingSlot pickingSlot)
 			return;
 		}
 
+		final boolean isSlotAllocatedByAConnectedComponent = SpringContextHolder.instance
+				.getBeansOfType(PickingSlotConnectedComponent.class)
```

Comment:
i suggest introducing getPickingSlotConnectedComponents() method.

In that method we can use a class field to lazy initialize the List of PickingSlotConnectedComponents,
so we won't search it in spring context each time (not sure how slow that is)

Reply:
or... u can introduce PickingSlotConnectedComponentRegistry spring component class (feel free to find a better name)  that acts a facade and dispatches requests to all PickingSlotConnectedComponent beans.